### PR TITLE
fix: add Vendetta social links in Settings

### PR DESCRIPTION
**Pull request by danilwhale**
use "Vendetta/Sunset" instead of "Sunset" inside unproxied plugin installation warning
also add links to vendetta server/github 🥺

### CODE REVIEW

File: src/lib/constants.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 export const DISCORD_SERVER = "https://discord.gg/tNPPbcPU2T";
+export const VENDETTA_DISCORD_SERVER = "https://discord.gg/vendetta-mod";
```
Comment (Ushie) [inline src/lib/constants.ts:2]: Use a permanent invite URL, the vanity will likely be lost due to the server archival

File: src/ui/settings/data.tsx [hunk 2]
```diff
@@ -51,7 +51,7 @@ export const getScreens = (youKeys = false): Screen[] => [
                         if (!input.startsWith(PROXY_PREFIX) && !settings.developerSettings)
                             setImmediate(() => showConfirmationAlert({
                                 title: "Unproxied Plugin",
-                                content: "The plugin you are trying to install has not been proxied/verified by Sunset staff. Are you sure you want to continue?",
+                                content: "The plugin you are trying to install has not been proxied/verified by Vendetta/Sunset staff. Are you sure you want to continue?",
```
Comment (Ushie) [inline src/ui/settings/data.tsx:54]: This is Sunset, Vendetta cannot verify plugins on our behalf

File: src/ui/settings/pages/General.tsx [hunk 3]
```diff
@@ -98,6 +98,21 @@ export default function General() {
...
+                        trailing={FormRow.Arrow}
+                        onPress={() => url.openDeeplink(VENDETTA_DISCORD_SERVER)}
+                    />
+                    <FormDivider />
+                    <FormRow
+                        label="GitHub"
+                        leading={<FormRow.Icon source={getAssetIDByName("img_account_sync_github_white")} />}
+                        trailing={FormRow.Arrow}
+                        onPress={() => url.openURL(VENDETTA_GITHUB)}
+                    />
```
Comment (Ushie) [inline src/ui/settings/pages/General.tsx:114]: Instead of duplicating this, forEach would be more pleasant

**Review by Ushie — CHANGES_REQUESTED** [review]

File: src/lib/constants.ts (hunk 1 above)
Comment (danilwhale) [reply src/lib/constants.ts:2]: i dont know permanent url, #info channel is locked, so i used vendetta-mod

Comment (riivx) [reply src/lib/constants.ts:2]: > i dont know permanent url

simply get it from the org. it's https://discord.gg/n9QQ4XhhJP btw

**Review by riivx — APPROVED** [review]
looks great, i approve

**Comment by danilwhale** [comment]
image

**Comment by mufaroxyz** [comment]
LGTM

File: src/ui/settings/pages/General.tsx [hunk 4]
```diff
@@ -11,6 +11,11 @@ import Version from "@ui/settings/components/Version";
 const { FormRow, FormSwitchRow, FormSection, FormDivider } = Forms;
 const debugInfo = getDebugInfo();
 
+const links = [
+    { discord: DISCORD_SERVER, github: GITHUB, whose: "Sunset" },
```
Comment (Ushie) [inline src/ui/settings/pages/General.tsx:15]: "whose" is a bad key name, use something more semantically correct

something like this:

```json
[
    "Sunset": {
        discord: DISCORD_SERVER,
        github: GITHUB
   },
    "Vendetta": {
        ....
   }
]
```

Or
```json
[
    {
        name: "Sunset",
        ...
    },
    {
        name: "Vendetta",
        ....
    }
]
```

Excuse the formatting, I'm on mobile atm

**Review by Ushie — CHANGES_REQUESTED** [review]

**Review by Ushie — CHANGES_REQUESTED** [review]
Improve the PR title to represent what it actually does

**Comment by riivx** [comment]
LGTM

File: src/ui/settings/pages/General.tsx [hunk 5]
```diff
@@ -83,21 +88,26 @@ export default function General() {
...
-                    />
-                    <FormDivider />
-                    <FormRow
-                        label="GitHub"
-                        leading={<FormRow.Icon source={getAssetIDByName("img_account_sync_github_white")} />}
-                        trailing={FormRow.Arrow}
-                        onPress={() => url.openURL(GITHUB)}
-                    />
-                </FormSection>
+                {links.map((link) => (
```
Comment (Ushie) [inline src/ui/settings/pages/General.tsx:91]: Why map? you could forEach

**Review by Ushie — CHANGES_REQUESTED** [review]

File: src/ui/settings/pages/General.tsx (hunk 5 above)
Comment (danilwhale) [reply src/ui/settings/pages/General.tsx:91]: isnt map easier?

Comment (mufaroxyz) [reply src/ui/settings/pages/General.tsx:91]: For more reference: map returns the iterated values while forEach does not return to the scope it is called in. Hence why it is required to use map in JSX

https://react.dev/learn/rendering-lists

Comment (riivx) [reply src/ui/settings/pages/General.tsx:91]: then i guess it's resolved

Comment (mufaroxyz) [reply src/ui/settings/pages/General.tsx:91]: > then i guess it's resolved

I think so. You can merge that I guess

**Review by Ushie — APPROVED** [review]

**Comment by Ushie** [comment]
The PR should specify that this is about, settings page

**Comment by riivx** [comment]
changed the title, thanks for PR!

**Pull request merged**